File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcAddress/IfcAddress.vb
Public Class IfcAddress
    Public Purpose As IfcAddressTypeEnum
    Public Description As IfcText
    Public UserDefinedPurpose As IfcLabel

    ' === EXTENDED BY ===
    ' IfcPostalAddress
    ' IfcTelecomAddress

    ' === INVERSE CLAUSES ===
    ' OfPerson : SET [0:?] OF IfcPerson FOR Addresses
    ' OfOrganization : SET [0:?] OF IfcOrganization FOR Addresses

    ' === WHERE CLAUSES ===
    ' WR1 : (NOT(EXISTS(Purpose))) OR ((Purpose <> IfcAddressTypeEnum.USERDEFINED) OR ((Purpose = IfcAddressTypeEnum.USERDEFINED) AND EXISTS(SELF.UserDefinedPurpose)))
End Class

File: src/main/resources/output/IFC4X3_js/Entities/IfcAddress/IfcAddress.js
class IfcAddress {
    constructor() {
        /** @type {IFCADDRESSTYPEENUM} */
        this.Purpose = null;
        /** @type {IFCTEXT} */
        this.Description = null;
        /** @type {IFCLABEL} */
        this.UserDefinedPurpose = null;
    }

    // === EXTENDED BY ===
    // IfcPostalAddress
    // IfcTelecomAddress

    // === INVERSE CLAUSES ===
    // OfPerson : SET [0:?] OF IfcPerson FOR Addresses
    // OfOrganization : SET [0:?] OF IfcOrganization FOR Addresses

    // === WHERE CLAUSES ===
    // WR1 : (NOT(EXISTS(Purpose))) OR ((Purpose <> IfcAddressTypeEnum.USERDEFINED) OR ((Purpose = IfcAddressTypeEnum.USERDEFINED) AND EXISTS(SELF.UserDefinedPurpose)))
}
